### PR TITLE
🐛 Properly handle link actions for invalid deep links

### PR DESCRIPTION
**Pull request by andretortolano**
bug found during exploration test for the 3.1.5 release, its very low severity since its an edge case, but for future versions we can handle invalid deeplinks, we can also display a log error in the debugger so customers can figure out why they are not navigating properly

image

### CODE REVIEW

File: appcues/src/main/java/com/appcues/action/appcues/LinkAction.kt [hunk 1]
```diff
@@ -42,10 +46,12 @@ internal class LinkAction(
 
         if (isEligibleCustomTab) {
             linkOpener.openCustomTabs(uri)
-        } else {
+        } else try {
             // this will handle any in-app deep link scheme URLs OR any web urls that were
             // requested to open into the external browser application
             appcues.navigationHandler?.navigateTo(uri) ?: linkOpener.startNewIntent(uri)
+        } catch (exception: ActivityNotFoundException) {
```
Comment (iujames) [inline appcues/src/main/java/com/appcues/action/appcues/LinkAction.kt:53]: should it catch any Exception at this level? The `ActivityNotFoundException` is specific to the `startNewIntent` implementation, but in theory, if they have a custom `navigationHandler` registered, they may throw some other Exception type and we can guard against that too?

**Review by iujames — APPROVED** [review]
Fix looks great, just a question on possible making more general exception handling.

Nice work adding tests

File: appcues/src/main/java/com/appcues/action/appcues/LinkAction.kt (hunk 1 above)
Comment (andretortolano) [reply appcues/src/main/java/com/appcues/action/appcues/LinkAction.kt:53]: I would say if they do their own implementation they need to handle it themselves, at least here we can avoid generic exception

Comment (iujames) [reply appcues/src/main/java/com/appcues/action/appcues/LinkAction.kt:53]: Ok, sounds reasonable to me, thanks.
